src/sql/connection/connection.py

```diff
@@ -717,34 +719,40 @@ def _connection_execute(self, query, parameters=None):
             Parameters to use in the query (:variable format)
         """
         parameters = parameters or {}
-
         # we do not support multiple statements
         if len(sqlparse.split(query)) > 1:
             raise NotImplementedError("Only one statement is supported.")
 
-        words = query.split()
-
-        if words:
-            first_word_statement = words[0].lower()
-        else:
-            first_word_statement = ""
-
-        # NOTE: in duckdb db "from TABLE_NAME" is valid
-        # TODO: we can parse the query to ensure that it's a SELECT statement
-        # for example, it might start with WITH but the final statement might
-        # not be a SELECT
-        # `summarize` is added to support %sql SUMMARIZE table in duckdb
-        is_select = first_word_statement in {"select", "with", "from", "summarize"}
-
         operation = partial(self._execute_with_parameters, query, parameters)
         out = self._execute_with_error_handling(operation)
 
         if self._requires_manual_commit:
-            # calling connection.commit() when using duckdb-engine will yield
-            # empty results if we commit after a SELECT statement
-            # see: https://github.com/Mause/duckdb_engine/issues/734
-            if is_select and self.dialect == "duckdb":
-                return out
+            # Calling connection.commit() when using duckdb-engine will yield
+            # empty results if we commit after a SELECT or SUMMARIZE statement,
+            # see: https://github.com/Mause/duckdb_engine/issues/734.
+            if self.dialect == "duckdb":
```

Comment:
we've had some weird experiences when using sqlglot for parsing (we're actually planning to remove it but haven't done so)

you can use `sqlparse.parse` instead to replicate this functionality. sqlparse won't detect things like `SUMMARIZE df` so you might need to do some manual parsing but it's better than using sqlglot

some quick stuff I tried:

```python
In [1]: import sqlparse

In [2]: sqlparse.parse("select * from stuff")
Out[2]: (<Statement 'select...' at 0x107704200>,)

In [3]: p=sqlparse.parse("select * from stuff")[0]

In [4]: p.get_type()
Out[4]: 'SELECT'

In [5]: sqlparse.parse("select * from stuff")[0].get_type()
Out[5]: 'SELECT'

In [6]: sqlparse.parse("summarize")[0].get_type()
Out[6]: 'UNKNOWN'

In [7]: sqlparse.parse("with x as (select * from y) select * from x")[0].get_type()
Out[7]: 'SELECT'

In [8]: sqlparse.parse("from stuff")[0].get_type()
Out[8]: 'UNKNOWN'

In [9]: sqlparse.parse("from stuff")
Out[9]: (<Statement 'from s...' at 0x1075773E0>,)

In [10]: sqlparse.parse("summarize stuff")
Out[10]: (<Statement 'summar...' at 0x107AFD700>,)

In [11]: p=sqlparse.parse("summarize stuff")[0]

In [12]: p
Out[12]: <Statement 'summar...' at 0x10864FCA0>

In [13]: p.get_real_name()

In [14]: p.token_first()
Out[14]: <Identifier 'summar...' at 0x10864F5A0>

In [15]: token=p.token_first()

In [16]: token
Out[16]: <Identifier 'summar...' at 0x10864F5A0>

In [17]: token.value
Out[17]: 'summarize stuff'

In [18]:

```